friends/src/App.js

```diff
@@ -22,21 +25,46 @@ class App extends Component {
         });
   }
 
-  //FORM HANDLER
-  formHaner(event) {
-    event.preventDefault();
-    
+  //SAVE FRIEND
+  saveFriend = (friends) => {
+    axios.post('http://localhost:5000/friends', friends)
+    .then((res) => {
+      console.log(res);
+      this.setState({
+          friends: res.data,
+      })
+      alert('Friend Updated');
+    }) 
+    .catch((err) => {
+      console.log('you fucked up because', err);
+    })
   }
 
-  saveFriend(name, age, email) {
-    axios.post('http://localhost:5000/friends', {
-      name: name,
-      age: age,
-      email: email,
+  //REMOVE FRIEND 
+  deleteFriend = (id) => {
+    axios.delete(`http://localhost:5000/friends/${id}`)
+    .then((res) => {
+      this.setState({
+          friends: res.data,
+        })
+      })
+    .catch((err) => {
+      console.log('you fucked up because', err);
     })
+  }
+
+  //UPDATE FRIEND 
+  updateFriend = (id, update) => {
+    axios.put(`http://localhost:5000/friends/${id}`, update)
     .then((res) => {
       console.log(res);
-    }) 
+      if(update.name === '' || update.age === '' || update.email === '') {
```

Comment:
You can use the required attribute in the input field to check if there is empty input